hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/web/WebHdfsFileSystem.java

```diff
@@ -307,6 +310,20 @@ public StorageStatistics provide() {
                 return new DFSOpsCountStatistics();
               }
             });
+    pathPrefix = PATH_PREFIX;
+    boolean useBasePath = conf.getBoolean(
+        HdfsClientConfigKeys.DFS_CLIENT_WEBHDFS_USE_BASE_PATH_KEY,
+        HdfsClientConfigKeys.DFS_CLIENT_WEBHDFS_USE_BASE_PATH_DEFAULT
+    );
+    if (uri.getPath() != null && !uri.getPath().equals("") && useBasePath) {
```

Comment:
although uri can not be null, but better to have null-check on that.

Reply:
I've added a `uri != null` check 👍